deno_dist/request.ts

```diff
@@ -69,14 +69,14 @@ export class HonoRequest<P extends string = '/', I extends Input['out'] = {}> {
   header(name: string): string | undefined
   header(): Record<string, string>
   header(name?: string) {
-    const headerData: Record<string, string> = {}
+    const headerData: Record<string, string | undefined> = {}
```

Comment:
Without this, I'd have to add `as string | unknown` to line 79, which is ugly. Ideally we'd turn on [`"noUncheckedIndexedAccess": true`](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess) in the tsconfig, but that makes _many_ things break so I consider it out of scope for this PR.